mesh/pmesh.cpp

```diff
@@ -103,13 +103,13 @@ ParMesh& ParMesh::operator=(ParMesh &&mesh)
    return *this;
 }
 
-ParMesh::ParMesh(MPI_Comm comm, Mesh &mesh, int *partitioning_,
+ParMesh::ParMesh(MPI_Comm comm, Mesh &mesh, const int *partitioning_,
                  int part_method)
    : glob_elem_offset(-1)
    , glob_offset_sequence(-1)
    , gtopo(comm)
 {
-   int *partitioning = NULL;
+   const int *partitioning = partitioning_ ? partitioning_ : nullptr;
```

Comment:
I think we can eliminate this line and rename `partitioning_` to `partitioning`.

Reply:
Discussion continued in https://github.com/mfem/mfem/pull/4190#discussion_r1607117726.